test/int/cluster_test.go

```diff
@@ -112,15 +112,111 @@ var _ = Describe("AtlasCluster", func() {
-			1200, interval).Should(BeTrue())
+			timeout, interval).Should(BeTrue())
 
 		lastGeneration++
 	}
 
+	Describe("Create cluster & change ReplicationSpecs", func() {
+		It("Should Succeed", func() {
+			expectedCluster := mdbv1.DefaultGCPCluster(namespace.Name, createdProject.Name)
+
+			By(fmt.Sprintf("Creating the Cluster %s", kube.ObjectKeyFromObject(expectedCluster)), func() {
+				createdCluster.ObjectMeta = expectedCluster.ObjectMeta
+				Expect(k8sClient.Create(context.Background(), expectedCluster)).ToNot(HaveOccurred())
+
+				Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterCreatingFunc()),
+					30*time.Minute, interval).Should(BeTrue())
+
+				doCommonChecks()
+				checkAtlasState()
+			})
+
+			By("Updating ReplicationSpecs", func() {
+				createdCluster.Spec.ReplicationSpecs = append(createdCluster.Spec.ReplicationSpecs, mdbv1.ReplicationSpec{
+					NumShards: int64ptr(2),
+				})
+				createdCluster.Spec.ClusterType = "SHARDED"
+				performUpdate(40 * time.Minute)
+				doCommonChecks()
+				checkAtlasState()
+			})
+		})
+	})
+
+	Describe("Create cluster & increase DiskSizeGB", func() {
+		It("Should Succeed", func() {
+			expectedCluster := mdbv1.DefaultGCPCluster(namespace.Name, createdProject.Name)
+
+			By(fmt.Sprintf("Creating the Cluster %s", kube.ObjectKeyFromObject(expectedCluster)), func() {
+				createdCluster.ObjectMeta = expectedCluster.ObjectMeta
+				Expect(k8sClient.Create(context.Background(), expectedCluster)).ToNot(HaveOccurred())
+
+				Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterCreatingFunc()),
+					1800, interval).Should(BeTrue())
+
+				doCommonChecks()
+				checkAtlasState()
+			})
+
+			By("Increasing InstanceSize", func() {
+				createdCluster.Spec.ProviderSettings.InstanceSizeName = "M30"
+				performUpdate(40 * time.Minute)
+				doCommonChecks()
+				checkAtlasState()
+			})
+		})
+	})
+
+	Describe("Create cluster & change it to GEOSHARDED", func() {
+		It("Should Succeed", func() {
+			expectedCluster := mdbv1.DefaultGCPCluster(namespace.Name, createdProject.Name)
+
+			By(fmt.Sprintf("Creating the Cluster %s", kube.ObjectKeyFromObject(expectedCluster)), func() {
+				createdCluster.ObjectMeta = expectedCluster.ObjectMeta
+				Expect(k8sClient.Create(context.Background(), expectedCluster)).ToNot(HaveOccurred())
+
+				Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterCreatingFunc()),
+					1800, interval).Should(BeTrue())
+
+				doCommonChecks()
+				checkAtlasState()
+			})
+
+			By("Change cluster to GEOSHARDED", func() {
+				createdCluster.Spec.ClusterType = "GEOSHARDED"
+				createdCluster.Spec.ProviderSettings.RegionName = ""
+				createdCluster.Spec.ReplicationSpecs = []mdbv1.ReplicationSpec{
```

Comment:
[q] does this deploy 3 nodes in east_us and 0 - in west_us?

Reply:
You're right - should I change it to 2/1 instead of 3/0?

Reply:
yes, 2/1 looks good to me
What do you think if we add 1 readonly/analytics node to some region?